File: observer/java/Waiter.java
package observer.java;

public class Waiter extends Observer {

    private Chef chef;
    public Waiter(Chef chef) {
        this.chef = chef;
        this.chef.attach(this);
    }

    public void stopObserving() {
        chef.detach(this);
    }

    @Override
    public void update(String message) {
        System.out.println("Waiter received message: " + message);
    }
}

File: observer/golang/waiter.go
package main

import "fmt"

// Waiter is a concrete observer.
type Waiter struct {
	Name string
}

func (w *Waiter) Update(orderName string) {
	fmt.Printf("Waiter %s: Picking up %s to serve to table.\n", w.Name, orderName)
}
